src/index.ts

```diff
@@ -150,11 +150,24 @@ export function disableExperimentalFragmentVariables() {
   experimentalFragmentVariables = false;
 }
 
-export default Object.assign(gql, {
+const extras = {
   gql,
-  default: gql,
   resetCaches,
   disableFragmentWarnings,
   enableExperimentalFragmentVariables,
   disableExperimentalFragmentVariables,
-});
+};
+
+export namespace gql {
+  export const {
+    gql,
+    resetCaches,
+    disableFragmentWarnings,
+    enableExperimentalFragmentVariables,
+    disableExperimentalFragmentVariables,
+  } = extras;
+}
+
+gql.default = gql;
```

Comment:
Unfortunately, since `default` is a reserved word, there doesn't seem to be any way to export it from the `namespace gql` block above.

Reply:
Waiting on https://github.com/microsoft/TypeScript/issues/24164